.changeset/beige-roses-cross.md

```diff
@@ -0,0 +1,6 @@
+---
+'@firebase/data-connect': patch
+---
+
+- Add check for non-query refType in `executeQuery` that throws error on refType mismatch
+- Add unit test to check that error is thrown when `executeQuery` receives non-query refType, and not thrown otherwise
```

Comment:
Changelogs get added to release notes, and the unit tests aren't really relevant. Can we say something like:
`Throw error when calling `executeQuery` with mutations`